tinyusb: Update to the recent master.

robert-hh opened the pull request:
This update requires changes to the rp2, nrf and mimxrt ports.
    
- rp2: change tud_task() into tud_task_ext() in mpconfigport.h
- nrf: include device/usbd.h in mpconfigport.h, for a backward compatible use of tud_task().
- mimxrt: use
    lib/tinyusb/src/portable/chipidea/ci_hs/dcd_ci_hs.c instead of
    lib/tinyusb/src/portable/nxp/transdimension/dcd_transdimension.c
    and change a #define.
    
The topics I know which are fixed with this version:
- fix the SAMD USB lock-up. (issue  #8521).
- support the MIMXRT11XX series of MCUs.
- fix a wrong pin definition for MIMXRT1050_EVKB.

Tested with the MIMXRT boards, Pico RP2, SAMD boards. Builds with the nrf port.

dpgeorge commented:
Thanks, this looks good.  I tested on a Pico and it works fine.

It looks like TinyUSB has added a few commits since yours, for better RP2040 support.  So maybe worth getting the very latest master commit of it?

robert-hh commented on ports/rp2/mpconfigport.h:
```diff
@@ -220,7 +220,7 @@ extern void mp_thread_end_atomic_section(uint32_t);
 
 #if MICROPY_HW_ENABLE_USBDEV
-#define MICROPY_HW_USBDEV_TASK_HOOK extern void tud_task(void); tud_task();
+#define MICROPY_HW_USBDEV_TASK_HOOK extern void tud_task_ext(uint32_t, bool); tud_task_ext(1000, false);
```
The #define by tinyusb uses max_int. But since it is not used, the number does not matter.
Besides that: Tested on a Arduino Nano 33 BLE as well.

robert-hh replied:
Done

robert-hh committed:
lib/tinyusb: Update to the most recent master.

robert-hh committed:
lib/tinyusb: Code adaption after the update.

Changes are required to the rp2, nrf and mimxrt ports.

rp2: change tud_task() into tud_task_ext()
mimxrt: use
    lib/tinyusb/src/portable/chipidea/ci_hs/dcd_ci_hs.c instead of
    lib/tinyusb/src/portable/nxp/transdimension/dcd_transdimension.c
nrf: add a definition for the changed tud_task().
    tud_task() is changed into tud_task_ext(), and the #define for
    backward compatibility is in src/device/usbd.h.

The items I know which are fixed with this version:
- fix to the SAMD USB lock-up.
- support the MIMXRT11XX series of MCUs.
- Fix a wrong pin definition for MIMXRT1050_EVKB.

Tested with the MIMXRT boards, Pico RP2, SAMD boards, nrf Board.

dpgeorge commented:
Thanks for updating.  Merged in d1ef3815e6041c4fd569bb19f88080a0c55b20c6 and 80339f1a33faadfe85d0276a2e22701ec9a5b82e
